docs/guides/universal-receiver-delegate/set-default-implementation.md

```diff
@@ -79,37 +310,476 @@ const addressPermissionsNewArrayLengthHex = web3.utils.padLeft(
 );
 
 // bytes16 index `addressPermissionsOldArrayLengthHex` will serve as index
-const URDIndexInArrayHex = addressPermissionsOldArrayLengthHex.substring(
+const newElementIndexInArrayHex = addressPermissionsOldArrayLengthHex.substring(
   34,
   66,
 );
 
-// encode setData Payload on the Vault
-const setDataPayload = await myUP.methods['setData(bytes32[],bytes[])'](
-  [
+const dataKeys = [
+  URD_DATA_KEY,
+  ERC725YDataKeys.LSP6['AddressPermissions[]'].length,
+  ERC725YDataKeys.LSP6['AddressPermissions[]'].index +
+    newElementIndexInArrayHex,
+  ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+    universalProfileURDAddress.substring(2),
+];
+const dataValues = [
+  universalProfileURDAddress,
+  addressPermissionsNewArrayLengthHex,
+  universalProfileURDAddress,
+  PERMISSIONS.SUPER_SETDATA,
+];
+```
+
+  </TabItem>
+
+  <TabItem value="ethersjs" label="ethers.js">
+
+```typescript title="Encode Data Keys & Values for updating the URD and its permissions"
+const addressPermissionsOldArrayLengthHex = await myUP['getData(bytes32)'](
+  ERC725YDataKeys.LSP6['AddressPermissions[]'].length,
+);
+
+const addressPermissionsNewArrayLength =
+  ethers.toNumber(addressPermissionsOldArrayLengthHex) + 1;
+
+const addressPermissionsNewArrayLengthHex = ethers.toBeHex(
```

Comment:
I am not sure which function from ethers is this.
I could not find in it in the etherjs docs.

https://docs.ethers.org/v5/search/?search=toBeHex

Reply:
Yeah my bad, it was taken from ether v6

Reply:
https://docs.ethers.org/v6-beta/api/utils/#toBeHex

Reply:
I am not sure if we should put functions from the beta version of ethers v6.

Also because at the beginning of the guide we say to install with 

```
npm install ethers ...
```

But the beta version of ethers is installed via `npm install ethers@beta`.

Leave it to @CallumGrindle and @Hugoo to decide

Reply:
Well, I guess v6 will be more "bulletproof"